Added ReadToEnd() function

jeffwork2021 opened the pull request:
This is my first Pull Request and not sure I understand the problem well. Hope this helps.

jeffwork2021 committed:
Added ReadToEnd() function

jeffwork2021 committed:
Update SubstringReader.cs

jeffwork2021 reviewed (commented):
Fixed the expression body for method issue

tacosontitan commented:
@jeffwork2021 thank you for your contribution!

I apologize for the delay in reviewing your changes, I've been sick recently. There are a few things still needed to consider the issue complete:

- [ ] `public` methods should be defined prior to `protected` or `private` methods.
- [ ] `ReadToEnd<T>` should also offer an overload that requires an `ISerializationProvider` so that consumers have more than one way to parse the result.
- [ ] Both overloads should move the current position of the reader to the end of the source string to keep consistent with the other `Read` methods.

jeffwork2021 committed:
Update SubstringReader.cs

Reordered ReadtoEnd() methods before protected and private methods. Added additional version to support additional ISerialization parameter.

jeffwork2021 commented:
Take good care and wish you speedy recovery!
Just update the source again and hope this does not make the commit history too messy. Please let me know if another request would be better.
